### ISSUE TITLE
Host CPU usage is very high while runing a cli image

### ISSUE THREAD

**Issue by lynus** [cloudius-systems/osv#914]
My host machine has one 8-core intel i7 processor. When running a standard cli image using run.py with no additional parameters, the host CPU usage is suspeciously high. Top utility reports the number is around 30% to 60%.
Meanwhile, I tested a vallina alpine image with qemu-system-x86_64 with kvm enabled, the number is always lower than 1%.
I also downloaded the cli image for kvm provided by osv.io page, The number is lower, but still relatively high, around 14%.
I wonder is it expected or something is wrong.

**Comment by nyh** [comment]
On my machine, running the CLI on an 8-core VM only shows 6% in "top". This is the sum of all 8 cores. I would have liked to see a lower number, but sadly we can't right now. There may be several things adding to this number:

1.  we have one thread running on 0.5% of one CPU ("page-access-scanner") looking for accessed pages.
2.  We have various utility threads (e.g., load balancing) that keep getting switched even if there's nothing interesting to do.
3. We have timer interrupts which happen even if there's nothing interesting to do.
4. When virtio-rng is enabled, we have a thread which constantly asks for entropy from the host and feeds it into the guest. This causes the host to constantly do work which may (?) be accounted to the qemu. 
5.  Finally, you can see in core/sched.cc's cpu::do_idle() that every time before the CPU goes idle, we spin a bit (currently 10000 times) looking for more work, to avoid slow IPIs.

One thing you can easily try is to reduce the number 10000 mentioned above to a lower number and see if it helps.
Another thing you can easily try is to comment out the "virtio-rng-pci" line in run.py, and see if that helps.

I don't know why you see 14% or even 30% or 60%, whereas I only see 6% (on the latest master branch). If you run "run.py --api" you can try "scripts/top.py" or scripts/freq.py (for various counters) to see if you see anything suspious taking a lot of CPU on the guest side.

**Comment by lynus** [comment]
It turns out the image that had high CPU usage was built by DEBUG mode. The release mode image is around 10% usage.
I also tested various spin number that you metioned. Here is the resulting CPU usage:

With  DEBUG mode image
10         100       1000        10000
11%       12%       17%       40-50%

With  release mode image
10              100               1000         10000
10-11%      10-11%         10-11%      11%

It seems the compiler has done greate job optimizing the code.
Another interesting obervation is that with 'top -H' command, I noticed only first two qemu-system-x86_64 threads has large cpu usage number while the rest nearly zero. Since all cpu has to do_idle, I can't  explain why other threads would not consume cpu as the first do.

**Comment by wkozaczuk** [comment]
Are there any concrete issues/improvements in this issue we can address? Or maybe rephrase the title? Is CPU utilization still around 10% in idle mode with latest version of OSv?
